app/assets/javascripts/discourse/app/components/modal/second-factor-add-security-key.js

```diff
@@ -0,0 +1,158 @@
+import {
+  bufferToBase64,
+  isWebauthnSupported,
+  stringToBuffer,
+} from "discourse/lib/webauthn";
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+
+export default class SecondFactorAddSecurityKey extends Component {
+  @service capabilities;
+  @service modal;
+
+  @tracked loading = false;
+  @tracked errorMessage = null;
+  @tracked securityKeyName;
+
+  get webauthnUnsupported() {
+    return !isWebauthnSupported();
+  }
+
+  @action
+  onClose() {
+    this.args.model.onClose();
+    this.modal.close();
+  }
```

Comment:
```suggestion
  @action
  onClose() {
    this.args.model.onClose();
    this.args.closeModal();
  }
```
(and then remove the modal service injection)

Or even better... how about getting rid of the `onClose()` and using `await` instead?